packages/hash/external-services/docker-compose.yml

```diff
@@ -44,19 +48,46 @@ services:
     ports:
       - "${POSTGRES_PORT}:5432"
     volumes:
-      - ../../../var/external-services/postgres/data:/data
+      - hash-postgres-data:/var/lib/postgresql/data
       - ./postgres/postgresql.conf:/etc/postgresql/postgresql.conf:ro
-      - ./postgres/init-user-db.sh:/docker-entrypoint-initdb.d/init-user-db.sh
+      - ./postgres/init-user-db.sh:/docker-entrypoint-initdb.d/init-user-db.sh:ro
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready --username ${POSTGRES_USER}"]
+      interval: 1s
+      timeout: 1s
+      retries: 5
+    command: -c 'config_file=/etc/postgresql/postgresql.conf'
+
+  graph:
+    init: true
+    depends_on:
+      postgres:
+        condition: service_healthy
+    build:
+      dockerfile: deployment/graph/Dockerfile
+      context: ../../graph
+      args:
+        PROFILE: dev
+    env_file:
+      - ../../../.env
+      - ../../../.env.local
```

Comment:
How do we want to be explicit here? (This is one of the reasons why I really dislike global `.env` files and pulling everything from there) Currently, we don't specify `.env.local` at all when invoking `docker-compose`, so it's currently not really possible to provide custom configurations. It would also be hard to capture all environment variables used in the Graph API (specified in the CLI arguments), e.g. `HASH_LOG_FORMAT`. 
How do we want to pass `HASH_KRATOS_PG_USER` as `POSTGRES_USER`, setting it explicitly in the `env:` section?

Reply:
> How do we want to be explicit here? 

Similar to how I've done it for the configuration of the `kratos` container below, so something like:

```suggestion
    env_file:
      - POSTGRES_USER=${HASH_GRAPH_PG_USER}
      - POSTGRES_PASSWORD=${HASH_GRAPH_PG_PASSWORD}
      - ...
```

This helps makes it explicit which environment variables the container is consuming from the `.env` file. If any are missing, docker-compose logs a warning that an environment variable is missing.

Reply:
The problem here is that we also define environment variables in the Graph and it would be really annoying if you cannot specify environment variables just because it's in the docker container because it was forgotten to add it to this list. Being explicit about this would be a big maintenance task.

Reply:
As discussed off-GH, we ermove the `env_file:` section and specify it manually. This is a bit annoying when adding more environment variables for `hash_graph`, but should be more robust with large `.env` files.